src/App.tsx

```diff
@@ -1,24 +1,119 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
+
+import { ErrorNotification } from './components/ErrorNotification';
+import { FooterMenu } from './components/FooterMenu';
+import { ListOfTodos } from './components/ListOfTodos';
+import { Header } from './components/Header';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+import { getTodos } from './api/todos';
+
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { CustomError } from './types/CustomError';
+import { ActiveTodoData } from './types/ActiveTodoData';
+
+import { USER_ID } from './utils/fetchClient';
+import { initData } from './constants/initData';
+import { useError } from './utils/useError';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(initData.todos);
+  const [filter, setFilter] = useState<Filter>(initData.filter);
+  const [customError, setError]
+    = useError(initData.customError);
+  const [activeTodoData, setActiveTodo]
+    = useState<ActiveTodoData>(initData.activeTodoData);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(initData.tempTodo);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setActiveTodo({
+          hasActiveTodo: todosFromServer.some(todo => !todo.completed),
+          activeLeft: todosFromServer.filter(todo => !todo.completed).length,
+        });
```

Comment:
1. do less iterations
2. do you really need `hasActiveTodo`? Couldn't you just always check `activeLeft.length` ?
```suggestion
        setTodos(todosFromServer);
        const activeTodos = todosFromServer.filter(todo => !todo.completed);
        
        setActiveTodo({
          hasActiveTodo: !!activeTodos.length,
          activeLeft: activeTodos .length,
        });
```